architecture/secure_storage.rst

```diff
@@ -334,6 +334,53 @@ a ``TEE_RPC_RPMB_CMD`` message, thanks to the ``thread_rpc_cmd()`` function.
 Most RPMB requests and responses use the data frame format defined by the JEDEC
 eMMC specification. HMAC authentication is implemented here also.
 
+Security considerations
+=======================
+The RPMB partition in eMMC can not be accessed until a key has been programmed
+on the device: this is a one time action for the lifetime of the device. Once
+the key has been written on the eMMC controller, the controller uses it to
+authenticate requests.
+
+In order to be able to program this key, OP-TEE must be configured with
+``CFG_RPMB_WRITE_KEY=y``.
+
+.. warning::
+   Be aware that this configuration will send the RPMB key in clear to the
+   non-secure side that relays the RPMB key programming request to the eMMC
+   hardware device.
+
+OP-TEE can either embed a built-in RPMB key (``CFG_RPMB_TESTKEY=y``) or derive
+it from platform specific secrets (``CFG_RPMB_TESTKEY=n``). The former case
+might be useful during development while the later is recommended for production
+devices.
+
+Deriving the key from secrets avoids OP-TEE from having to store it in memory
+therefore reducing the attack surface; OP-TEE derives the RPMB key from an
+internal set that includes the eMMC serial number and more importantly the
+Hardware Unique Key (HUK).
+
+For this configuration to be effective, the Hardware Unique Key - a unique
+identifier for the particular instantiation of the SoC - must not be publicly
+accessible (please notice that not all platforms might be enforcing this
+requirement).
```

Comment:
A link to [architecture/porting_guidelines.html#hardware-unique-key](https://optee.readthedocs.io/en/latest/architecture/porting_guidelines.html#hardware-unique-key) would be nice.